src/lib.rs

```diff
@@ -509,98 +562,122 @@ impl FailPoint {
 /// Registry with failpoints configuration.
 type Registry = HashMap<String, Arc<FailPoint>>;
 
-#[derive(Debug, Default)]
-struct FailPointRegistry {
+/// Fail point registry. Threads bound to the same registry share the same
+/// failpoints configuration.
+#[derive(Debug, Default, Clone)]
+pub struct FailPointRegistry {
     // TODO: remove rwlock or store *mut FailPoint
-    registry: RwLock<Registry>,
+    registry: Arc<RwLock<Registry>>,
 }
 
-lazy_static::lazy_static! {
-    static ref REGISTRY: FailPointRegistry = FailPointRegistry::default();
-    static ref SCENARIO: Mutex<&'static FailPointRegistry> = Mutex::new(&REGISTRY);
+/// Generate a new failpoint registry. The new registry will inherit the
+/// global failpoints configuration.
+///
+/// Each thread should be bound to exact one registry. Threads bound to the
+/// same registry share the same failpoints configuration.
+pub fn new_fail_group() -> FailPointRegistry {
+    let registry = REGISTRY_GLOBAL.registry.read().unwrap();
+    let mut new_registry = Registry::new();
+    for (name, failpoint) in registry.iter() {
```

Comment:
I don't think it's a good idea. It becomes a mess when users try to remove configuration from global.

Reply:
Can you give some examples

Reply:
In fail-rs, `Global` and `Local` registry is independent manager. The only connection between them is that `Local` will inherit `Global`'s failpoint configuration. When the establishment of `Local` is completed, they will no longer have contact, so as long as this is clearly stated in our documentation, I don't think users will be confused.

Reply:
I suddenly thought of an idea. we can add a parameter to `new_fail_group` to let users decide whether to inherit the global configuration.

Reply:
In what circumstances will inheritance be necessary? It gets worse especially when `Local` and `Global` configurations are separated. This is not how inheritance is expected to work.